The explicit disable of tx_delay destroys the registers and 
the AR 8035 based ports of some lantiq xrx200 based devices
does not work anymore. So reverting to the last known working
linux version 4.19 with this patch.

--- a/drivers/net/phy/at803x.c
+++ b/drivers/net/phy/at803x.c
@@ -288,20 +288,20 @@ static int at803x_config_init(struct phy_
 	 */
 
 	if (phydev->interface == PHY_INTERFACE_MODE_RGMII_ID ||
-	    phydev->interface == PHY_INTERFACE_MODE_RGMII_RXID)
+	    phydev->interface == PHY_INTERFACE_MODE_RGMII_RXID) {
 		ret = at803x_enable_rx_delay(phydev);
-	else
-		ret = at803x_disable_rx_delay(phydev);
-	if (ret < 0)
-		return ret;
+		if (ret < 0)
+			return ret;
+	}
 
 	if (phydev->interface == PHY_INTERFACE_MODE_RGMII_ID ||
-	    phydev->interface == PHY_INTERFACE_MODE_RGMII_TXID)
+	    phydev->interface == PHY_INTERFACE_MODE_RGMII_TXID) {
 		ret = at803x_enable_tx_delay(phydev);
-	else
-		ret = at803x_disable_tx_delay(phydev);
+		if (ret < 0)
+			return ret;
+	}
 
-	return ret;
+	return 0;
 }
 
 static int at803x_ack_interrupt(struct phy_device *phydev)
